jena-rdfconnection/src/test/java/org/apache/jena/rdfconnection/TestRDFConnectionHTTPHeaders.java

```diff
@@ -0,0 +1,121 @@
+                assertEquals("d", qe.getAcceptHeaderConstructDataset());
+            }
+        }
+    }
+
+    @Test void test_headers_non_parsed_fallback() {
+        try (RDFConnection conn = RDFConnectionRemote.service("urn:dummy-service")
+            .parseCheckSPARQL(false)
+            .acceptHeaderQuery("f") // fallback
+            .acceptHeaderSelectQuery(null)
+            .acceptHeaderAskQuery(null)
+            .acceptHeaderGraph(null)
+            .acceptHeaderDataset(null)
+            .build()) {
+
+            try (QueryExecution qExec = conn.query("dummy query string")) {
+                QueryExecHTTP qe = (QueryExecHTTP)QueryExec.adapt(qExec);
+                assertEquals("f", qe.getAcceptHeaderSelect());
+                assertEquals("f", qe.getAcceptHeaderAsk());
+                assertEquals("f", qe.getAcceptHeaderDescribe());
+                assertEquals("f", qe.getAcceptHeaderConstructGraph());
+                assertEquals("f", qe.getAcceptHeaderConstructDataset());
+            }
+        }
+    }
+
+    @Test void test_headers_parsed_fallback() {
+        try (RDFConnection conn = RDFConnectionRemote.service("urn:dummy-service")
+            .acceptHeaderQuery("f") // fallback
+            .acceptHeaderSelectQuery(null)
+            .acceptHeaderAskQuery(null)
+            .acceptHeaderGraph(null)
+            .acceptHeaderDataset(null)
+            .build()) {
+
+            try (QueryExecution qExec = conn.newQuery()
+                    .query("SELECT * { ?s ?p ?o }")
+                    .build()) {
+                QueryExecHTTP qe = (QueryExecHTTP)QueryExec.adapt(qExec);
+                assertEquals("f", qe.getAcceptHeaderSelect());
+
+                assertEquals("f", qe.getAcceptHeaderAsk());
+                assertEquals("f", qe.getAcceptHeaderDescribe());
+                assertEquals("f", qe.getAcceptHeaderConstructGraph());
+                assertEquals("f", qe.getAcceptHeaderConstructDataset());
+            }
+        }
+    }
+
+    /**
+     * Setting an override-header on the builder should set all
+     * headers on the QueryExecHTTP instance to that override.
+     */
+    @Test
+    public void test_override_header() {
+        try (QueryExecutionHTTP qExec =
+                QueryExecutionHTTPBuilder.create()
+                .endpoint("urn:dummy-service")
+                .queryString("SELECT * { ?s ?p ?o }")
+                .acceptHeader("o") // Override header
+                .build()) {
+            QueryExecHTTP qe = (QueryExecHTTP)QueryExec.adapt(qExec);
+            assertEquals("o", qe.getAppProvidedAcceptHeader());
+            assertEquals("o", qe.getAcceptHeaderSelect());
+            assertEquals("o", qe.getAcceptHeaderAsk());
+            assertEquals("o", qe.getAcceptHeaderDescribe());
+            assertEquals("o", qe.getAcceptHeaderConstructGraph());
+            assertEquals("o", qe.getAcceptHeaderConstructDataset());
+        }
+    }
+}
```

Comment:
There is a missing test:

```
        .acceptHeader("o")
        .acceptHeaderAskQuery("a")
```
which I would expect to set just the ASK query type to "a" and leave everything else as "o"

Reply:
Added as `TestRDFConnectionHTTPHeaders.test_override_header_02`.